### PR TITLE
Rename almost all Ctx functions

### PR DESCRIPTION
**Pull request by lunny**

### CODE REVIEW

**Review by zeripath — APPROVED** [review]
My only comment would be that perhaps the functions that run things in their own transaction should gain a suffix of Tx or the like?

File: services/projects/projects.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	if err != nil {
+		return err
+	}
+	defer committer.Close()
+
+	if err := project_model.DeleteProjectByID(ctx, id); err != nil {
+		return err
+	}
+
+	return committer.Commit()
```
Comment (delvh) [inline services/projects/projects.go]:
```suggestion
	db.WithTx(ctx, func(ctx context.Context) error {
		return project_model.DeleteProjectByID(ctx, id)
	})
```

File: services/repository/repository.go [hunk 2]
```diff
@@ -125,3 +126,18 @@ func LinkedRepository(ctx context.Context, a *repo_model.Attachment) (*repo_mode
...
+	if err != nil {
+		return err
+	}
+	defer committer.Close()
+
+	if err := repo_model.ChangeCollaborationAccessMode(ctx, repo, uid, mode); err != nil {
+		return err
+	}
+
+	return committer.Commit()
```
Comment (delvh) [inline services/repository/repository.go]:
```suggestion
	db.WithTx(db.DefaultContext, func(ctx context.Context) error {
		return repo_model.ChangeCollaborationAccessMode(ctx, repo, uid, mode)
	})
```

File: modules/repository/create.go [hunk 3]
```diff
@@ -125,10 +125,10 @@ func CreateRepositoryByExample(ctx context.Context, doer, u *user_model.User, re
 			return fmt.Errorf("IsUserRepoAdmin: %w", err)
 		} else if !isAdmin {
 			// Make creator repo admin if it wasn't assigned automatically
-			if err = addCollaborator(ctx, repo, doer); err != nil {
+			if err = AddCollaborator(ctx, repo, doer); err != nil {
 				return fmt.Errorf("addCollaborator: %w", err)
```
Comment (delvh) [inline modules/repository/create.go]:
```suggestion
				return fmt.Errorf("AddCollaborator: %w", err)
```

File: services/projects/projects.go (hunk 1 above)
Comment (lunny) [reply services/projects/projects.go]: I removed this function and use `db.AutoTx` to handle it.

File: models/issues/comment.go [hunk 4]
```diff
@@ -780,7 +780,7 @@ func (c *Comment) LoadPushCommits(ctx context.Context) (err error) {
 }
 
 // CreateCommentCtx creates comment with context
```
Comment (KN4CK3R) [inline models/issues/comment.go]:
```suggestion
// CreateComment creates comment with context
```

File: services/pull/comment.go [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+			if err != nil {
+				return err
+			}
+			checkStack = append(checkStack, parentCommit.ID.String())
+		}
+	}
+	return nil
+}
+
+// getCommitsFromRepo get commit IDs from repo in between oldCommitID and newCommitID
```
Comment (KN4CK3R) [inline services/pull/comment.go]:
```suggestion
// getCommitIDsFromRepo get commit IDs from repo in between oldCommitID and newCommitID
```

**Review by delvh — APPROVED** [review]

**Comment by lunny** [comment]
make L-G-T-M work

**Pull request merged**